File: ecote/Disjoint-set/팀결성.js
let input = prompt()
  .split(" ")
  .map((v) => parseInt(v));
let n = input[0];
let m = input[1];

let parent = Array(n + 1).fill(0);
for (let i = 1; i < n + 1; i += 1) {
  parent[i] = i;
}
const findParent = (parent, x) => {
  if (parent[x] != x) {
    parent[x] = findParent(parent, parent[x]);
  }
  return parent[x];
};
const unionParent = (parent, a, b) => {
  a = findParent(parent, a);
  b = findParent(parent, b);
  a < b ? (parent[b] = a) : (parent[a] = b);
};
for (let _ = 0; _ < n + 1; _ += 1) {
  let input = prompt()
    .split(" ")
    .map((v) => parseInt(v));
  let sig = input[0];
  let a = input[1];
  let b = input[2];

  if (sig == 1) {
    if (findParent(parent, a) == findParent(parent, b)) {
      console.log("YES");
    } else {
      console.log("NO");
    }
  } else if (sig == 0) {
    unionParent(parent, a, b);
  }
}

File: ecote/Disjoint-set/팀결성.py
def find_parent(parent, x):
    if parent[x] != x:
        parent[x] = find_parent(parent, parent[x])
    return parent[x]

def union_parent(parent, a, b):
    a = find_parent(parent, a)
    b = find_parent(parent, b)
    if a < b:
        parent[b] = a
    elif b > a:
        parent[a] = b

n, m = map(int, input().split())
parent = [0] * (n + 1)
for i in range(1, n + 1):
    parent[i] = i

for _ in range(m):
    e, a, b = map(int, input().split())
    if e == 0:
        union_parent(parent, a, b)
    elif e == 1:
        if find_parent(parent, a) == find_parent(parent, b):
            print("yes")
        else:
            print("no")

"""
7 8
0 1 3
1 1 7
0 7 6
1 7 1
0 3 7
0 4 2
0 1 1
1 1 1
"""